app/components/ui/molecules/PeoplePicker.js

```diff
@@ -145,11 +146,71 @@ class PeoplePicker extends React.Component {
     this.setState({ searchValue })
   }
 
-  filterPeople(people) {
-    const searchValue = this.state.searchValue.toLowerCase()
-    return people.filter(person =>
-      person.name.toLowerCase().includes(searchValue),
-    )
+  filterPeople = (people, searchValue, field) => {
+    if (!searchValue) return people
+
+    const inputValue = searchValue.trim().toLowerCase()
+    if (!inputValue) return people
+
+    if (inputValue.split(' ').length > 1) {
+      const matches = people.filter(person =>
+        person[field].toLowerCase().includes(inputValue),
+      )
+      return matches.map(elem => {
+        const clonedElem = cloneDeep(elem)
+        clonedElem.matchIndex = elem[field].toLowerCase().indexOf(inputValue)
+        return clonedElem
+      })
+    }
+    return this.filterPeopleSingle(people, inputValue, field)
+  }
+
+  filterPeopleSingle = (people, inputValue, field) => {
+    let maxWords = 0
+    people.forEach(person => {
+      const numWords = person[field].match(/[^ ]+/g).length
+      if (numWords > maxWords) {
+        maxWords = numWords
+      }
+    })
+
+    let j
```

Comment:
Where do we initialise `j` ?

Reply:
This `j` made more sense when I had the filtering/mapping functions in place but because of some weird eslint rules (which I didn't want to disable) now it looks all weird. Yeah it guess it's really hard to read like this.

I think a better question here is this: we have to see if the person matches the input value once, when we filter the results, and then again when we render the results because I need to know the index of the match so I can turn the matched part bold. Right now I perform this match test only once (here) and store the match index in the people list so I can use it again in the render function. Does it make more sense to test for the match twice (here and in the render function)? I think so, I can probably put this part in a function and use it in both places. We don't save that much by performing the match test twice since it's just a small list of items. It will be easier to read.

Reply:
I'll delete my other comment as it amounts to the same thing!

Reply:
@dryhten  ... Happy to do the merge if you are planning to do the re-factor? Know we've been busy today - just putting this here as a reminder ;)